quickstart/201-azfw-with-avzones/main.tf

```diff
@@ -0,0 +1,264 @@
+  location              = azurerm_resource_group.rg.location
+  computer_name         = "jumpbox"
+  size                  = var.virtual_machine_size
+  admin_username        = var.admin_username
+  admin_password        = random_password.password.result
+  network_interface_ids = [azurerm_network_interface.vm_jump_nic.id]
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+    disk_size_gb         = "128"
+  }
+  source_image_reference {
+    publisher = "MicrosoftWindowsServer"
+    offer     = "WindowsServer"
+    sku       = "2019-Datacenter"
+    version   = "latest"
+  }
+  boot_diagnostics {
+    storage_account_uri = azurerm_storage_account.sa.primary_blob_endpoint
+  }
+}
+
+resource "azurerm_firewall_policy" "azfw_policy" {
+  name                     = "azfw-policy"
+  resource_group_name      = azurerm_resource_group.rg.name
+  location                 = azurerm_resource_group.rg.location
+  sku                      = var.firewall_sku_tier
+  threat_intelligence_mode = "Alert"
+}
+
+resource "azurerm_firewall_policy_rule_collection_group" "prcg" {
+  name               = "prcg"
+  firewall_policy_id = azurerm_firewall_policy.azfw_policy.id
+  priority           = 300
+  application_rule_collection {
+    name     = "appRc1"
+    priority = 101
+    action   = "Allow"
+    rule {
+      name = "appRule1"
+      protocols {
+        type = "Http"
+        port = 80
+      }
+      protocols {
+        type = "Https"
+        port = 443
+      }
+      destination_fqdns = ["www.microsoft.com"]
+      source_addresses  = ["10.10.1.0/24"]
+    }
+  }
+  network_rule_collection {
+    name     = "netRc1"
+    priority = 200
+    action   = "Allow"
+    rule {
+      name              = "netRule1"
+      protocols         = ["TCP"]
+      source_addresses  = ["10.10.1.0/24"]
+      destination_addresses = [ "*" ]
+      destination_ports = ["8000", "8999"]
+    }
+  }
+}
+
+resource "azurerm_firewall" "fw" {
+  name                = "azfw"
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+  sku_name            = "AZFW_VNet"
+  sku_tier            = var.firewall_sku_tier
+  zones               = ["1", "2", "3"]
+  ip_configuration {
+    name                 = "azfw-ipconfig"
+    subnet_id            = azurerm_subnet.azfw_subnet.id
+    public_ip_address_id = azurerm_public_ip.pip_azfw.id
+  }
+  firewall_policy_id = azurerm_firewall_policy.azfw_policy.id
+}
+
+resource "azurerm_route_table" "rt" {
+  name                          = "rt-azfw-eus"
+  location                      = azurerm_resource_group.rg.location
+  resource_group_name           = azurerm_resource_group.rg.name
+  disable_bgp_route_propagation = false
+  route {
+    name                   = "azfwDefaultRoute"
+    address_prefix         = "0.0.0.0/0"
+    next_hop_type          = "VirtualAppliance"
+    next_hop_in_ip_address = "10.10.0.4"
```

Comment:
I'd like to try `azurerm_firewall.fw.ip_configuration[0].private_ip_address` instead of this hard coded address here.

Reply:
will give it a shot

Reply:
still works. fixed. thanks for the suggestion.